profiling/src/internal/backup_sample_types.rs

```diff
@@ -0,0 +1,115 @@
+// Copyright 2021-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::api::{Period, ValueType};
+use std::mem::ManuallyDrop;
+
+/// In order to reset the profile, it needs the sample types. However, it's
+/// inconvenient to reconstruct this because of differences in api and internal
+/// layouts. This struct serves as a backup of the original sample types, so
+/// we can just re-use them as needed. It does a single allocation for
+/// everything from the global allocator.
+pub struct BackupSampleTypesAndPeriod {
+    /// The static bounds are lies, they are tied to the `storage`. Only pass
+    /// out references with the same lifetime as &self.
+    sample_types: ManuallyDrop<Box<[ValueType<'static>]>>,
+    /// Despite our manual drop order, it is still recommended to order the
+    /// fields in the order things should be dropped.
+    storage: ManuallyDrop<Vec<(String, String)>>,
+    period: Option<((String, String), i64)>,
```

Comment:
Minor: Out of curiosity, it seems that often we either omit or pass in a dummy period. I  wonder if we should just get rid of this parameter for libdatadog API users (and libdatadog would always just insert a default one).

Reply:
Yeap, for sure, more of a passing observation.